server/icat/include/irods/icatHighLevelRoutines.hpp

```diff
@@ -447,4 +449,146 @@ auto chl_delay_rule_unlock(RsComm& _comm, const char* _rule_ids) -> int;
+/// associated.
+///
+/// \p auth_scheme indicates the name of the authentication scheme with which this session token is being used.
+///
+/// \p session_token is the value of the session token to check. This should be a 36-character UUID.
+/// \endparblock
+/// \param[out] _valid A pointer which will hold a value of 0 if the session token check fails, or 1 if it succeeds.
+///
+/// \retval 0 On success.
+/// \retval <0 \parblock
+/// If an error occurs while querying for the provided session token in the catalog. If the search yields no results
+/// for the given inputs, this is not an error. In that case, the \p _valid out pointer will hold a value of 0.
+/// \endparblock
+///
+/// \since 5.1.0
+auto chl_check_session_token(RsComm* _comm, const char* _json_input, int* _valid) -> int;
+
+/// \brief High-level wrapper for creating session tokens.
+///
+/// \param[in] _comm The communication object.
+/// \param[in] _json_input \parblock
+/// \code{.js}
+/// {
+///     "user_name": "<string>",
+///     "zone_name": "<string>",
+///     "auth_scheme": "<string>",
+///     "expires": <bool>
+/// }
+/// \endcode
+///
+/// \p user_name and \p zone_name are the user name and zone name of the user with whom the session token to be created
+/// should be associated.
+///
+/// \p auth_scheme indicates the name of the authentication scheme with which this session token should be used.
+///
+/// \p expires is an optional parameter. If specified and set to false, the new session token will never expire. If
+/// the option is not specified or set to true, the session token will expire. Note: Only rodsadmins are allowed to
+/// request session tokens which do not expire.
+/// \endparblock
+/// \param[out] _token A pointer which will hold the value of the newly created session token.
+///
+/// \retval 0 On success.
+/// \retval <0 On failure.
+///
+/// \since 5.1.0
+auto chl_make_session_token(RsComm* _comm, const char* _json_input, char** _token) -> int;
+
+/// \brief High-level wrapper for deleting session tokens.
+///
+/// \param[in] _comm The communication object.
+/// \param[in] _json_input \parblock
+/// A JSON string of the following form containing information about the session tokens to delete:
+/// \code{.js}
+/// {
+///     "user_name": "<string>",
+///     "zone_name": "<string>",
+///     "expired_only": <bool>
+/// }
+/// \endcode
+///
+/// If \p expired_only is true, only expired session tokens associated with the specified user will be deleted. Else,
+/// all session tokens associated with the specified user will be deleted.
+///
+/// \p user_name and \p zone_name are the user name and zone name of the user with whom the session tokens to be deleted
+/// are associated. If no \p user_name or \p zone_name are specified, session tokens belonging to any user will be
+/// deleted. Not specifying a user should be done deliberately and with caution.
```

Comment:
What happens if I provide `"user_name": "", zone_name": ""`?

Reply:
The DB operation takes them as an empty user name and an empty zone name, and tries to use them as bind variables. I imagine this will end with an error. It would be better if this was in a function that I could unit test, but as it stands, it is only accessible through the GeneralAdmin API (i.e. `iadmin`).

Reply:
> I think that would fall outside the scope of this PR. Maybe we can make an issue to look into that later, perhaps for iRODS 6.

That's fair.

> In the meantime, should we make an intentional decision on what happens in that situation? Should an empty username / zone name be treated the same way as a missing username / zone name?

Yes, we should be intentional when it comes to passwords and other sensitive operations. I think treating the two cases differently is good. If the username and/or zone are defined, the client is telling us they are targeting a specific user - i.e. empty strings lead to an error. If username and zone aren't defined, the client is targeting all users.